src/main/scala/org/broadinstitute/dsde/workbench/sam/google/DisableUsersMonitor.scala

```diff
@@ -0,0 +1,192 @@
+           ): Props =
+    Props(new DisableUsersMonitorActor(pollInterval, pollIntervalJitter, pubSubDao, pubSubSubscriptionName, userService))
+}
+
+class DisableUsersMonitorActor(
+  val pollInterval: FiniteDuration,
+  pollIntervalJitter: FiniteDuration,
+  pubSubDao: GooglePubSubDAO,
+  pubSubSubscriptionName: String,
+  userService: UserService
+) extends Actor with LazyLogging with FutureSupport {
+  import DisableUsersMonitor._
+  import context._
+
+  self ! StartMonitorPass
+
+  // fail safe in case this actor is idle too long but not too fast (1 second lower limit)
+  setReceiveTimeout(max((pollInterval + pollIntervalJitter) * 10, 1 second))
+
+  private def max(durations: FiniteDuration*): FiniteDuration = {
+    implicit val finiteDurationIsOrdered: FiniteDuration.FiniteDurationIsOrdered.type = scala.concurrent.duration.FiniteDuration.FiniteDurationIsOrdered
+    durations.max
+  }
+
+  override def receive: Receive = {
+    case StartMonitorPass =>
+      // start the process by pulling a message and sending it back to self
+      pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
+
+    case Some(message: PubSubMessage) =>
+      attemptToDisableUser(message) pipeTo self
+
+    case None =>
+      // there was no message to wait and try again
+      val nextTime = org.broadinstitute.dsde.workbench.util.addJitter(pollInterval, pollIntervalJitter)
+      system.scheduler.scheduleOnce(nextTime.asInstanceOf[FiniteDuration], self, StartMonitorPass)
+
+    case ReportMessage(disableUserResponse, ackId) =>
+      handleDisableUserResponse(disableUserResponse, ackId) pipeTo self
+
+    case FailToDisable(t, ackId) =>
+      t match {
+        case userNotFound: WorkbenchExceptionWithErrorReport if userNotFound.errorReport.statusCode.contains(StatusCodes.NotFound) =>
+          // this can happen if a user is deleted before the disable message is handled
+          // acknowledge it so we don't have to handle it again
+          acknowledgeMessage(ackId).map(_ => StartMonitorPass) pipeTo self
+          logger.info(s"user to disable not found: ${userNotFound.errorReport}")
+
+        case regrets: Throwable => throw regrets
+      }
+
+    case Status.Failure(t) => throw t
+
+    case ReceiveTimeout =>
+      throw new WorkbenchException("DisableUsersMonitorActor has received no messages for too long")
+
+    case x => logger.info(s"unhandled $x")
+  }
+
+  private def handleDisableUserResponse(disableUserResponse: DisableUserResponse, ackId: String) = {
+    import Tracing._
+    trace("DisableUsersMonitor-ReportMessage") { _ =>
+      disableUserResponse.value match {
+        case Some(_) =>
+          // If we have gotten to this point, the disableUser method has run, so we can assume the user is disabled
+          logger.info(s"disabled user: ${disableUserResponse.userId} ")
+        case None =>
+          logger.info(s"user to disable not found: ${disableUserResponse.userId}")
```

Comment:
If I am reading this right, if we end up here, things have probably gone wrong in a weird way, right?  The `attemptToDisableUser` claims that it succeeded, but there's something wrong with the Response object.  I don't know what we should do in this situation then.  Maybe just logging a message is fine and go ahead and ack the message.  Or maybe that's the wrong thing to do, maybe `attemptToDisableUser` thinks it succeeded, but it didn't and the user is still enabled?  I just want to make sure if we end up at this line of code, we understand the hows and whys and what it means and at the very least we're logging as meaningful a message as possible.

Reply:
This is what happens if `directoryDAO.loadUser(userId, samRequestContext)` returns None. So in that case, we are cool just logging that the user doesn't exist, and being happy. Because that means the user has been deleted or somesuch before we got to it.